lane.py

```diff
@@ -810,8 +810,9 @@ def works_for_specific_ids(self, _db, work_ids):
         qu = _db.query(mw).join(
             LicensePool, mw.license_pool_id==LicensePool.id
         ).filter(
-            mw.works_id.in_(work_ids)
-        )
+            mw.works_id.in_(work_ids),
+            LicensePool.work_id.in_(work_ids),
```

Comment:
I'm curious about this part - does the materialized view have license pools that don't match the works somehow?

Reply:
This allows the index to be used to limit the LicensePools before the `only_deliverable_works` criteria are applied. Otherwise it attempts to filter with that criteria against the whole licensepools table without using the index.

Reply:
:+1: